python/ray/dag/dag_operation_future.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, List, Optional, TypeVar, Union
+
+T = TypeVar("T")
+
+
+class DAGOperationFuture(ABC, Generic[T]):
+    @abstractmethod
+    def wait(self, waiter: Optional[Any] = None):
+        raise NotImplementedError
+
+
+class ReadyFuture(DAGOperationFuture):
```

Comment:
Add docs, maybe call it `ResolvedFuture`?

Reply:
Done